components/script/dom/htmlmediaelement.rs

```diff
@@ -1218,6 +1221,12 @@ impl HTMLMediaElement {
                 if self.ready_state.get() < ReadyState::HaveMetadata {
                     self.queue_dedicated_media_source_failure_steps();
                 }
+                if self.loop.get() {                    
```

Comment:
The step annotations above indicate that this may be the incorrect location to add the looping functionality. I suspect the concept of the end of the media resource may not even be implemented yet. @ferjm Am I correct in claiming so? If not, where do we handle the playback ended event?

Reply:
The concept of the [end of the media resource](https://html.spec.whatwg.org/multipage/media.html#ended-playback) is not implemented yet. It should be done as part of #22294.

I do believe this is the right place to add the looping functionality. But apart from seeking to the start position we need to do a couple of things more:

- Do an early return if the [previous condition](https://github.com/servo/servo/pull/22321/files#diff-aa93b0c055ef44fd49b191ba0d9046d9R1221) is `true`, as we don't want to loop if there are errors.
- Add a XXX comment saying that we need to identify if the stream is seekable or not to decide if we can loop. Please, reference #22297 in the comment.
- Set `playback_position` to `default_playback_start_position` here.